test/framework/src/main/java/org/elasticsearch/cluster/routing/TestShardRouting.java

```diff
@@ -28,193 +28,124 @@
 /**
  * A helper that allows to create shard routing instances within tests, while not requiring to expose
  * different simplified constructors on the ShardRouting itself.
+ *
+ * Please do not add more `newShardRouting`, consider using a aSharRouting builder instead
  */
 public class TestShardRouting {
 
-    public static ShardRouting newShardRouting(String index, int shardId, String currentNodeId, boolean primary, ShardRoutingState state) {
-        return newShardRouting(new ShardId(index, IndexMetadata.INDEX_UUID_NA_VALUE, shardId), currentNodeId, primary, state);
+    public static Builder aShardRouting(String index, int shardId, String currentNodeId, boolean primary, ShardRoutingState state) {
+        return new Builder(new ShardId(index, IndexMetadata.INDEX_UUID_NA_VALUE, shardId), currentNodeId, primary, state);
     }
 
-    public static ShardRouting newShardRouting(ShardId shardId, String currentNodeId, boolean primary, ShardRoutingState state) {
-        return newShardRouting(shardId, currentNodeId, primary, state, -1);
+    public static Builder aShardRouting(ShardId shardId, String currentNodeId, boolean primary, ShardRoutingState state) {
```

Comment:
Naming nit: we don't really call things `aSomethingOrOther` elsewhere, it seems kinda awkward to introduce that pattern here. I think `shardRouting` would be fine:

```suggestion
    public static Builder shardRouting(ShardId shardId, String currentNodeId, boolean primary, ShardRoutingState state) {
```